docs/QuickStart.md

```diff
@@ -0,0 +1,250 @@
+        test_acc = mnist_network.evaluate(mnist)
+        print ("Final result is", test_acc)
+
+if __name__ == '__main__':
+    params = {'data_dir': '/tmp/tensorflow/mnist/input_data', 'dropout_rate': 0.5, 'channel_1_num': 32, 'channel_2_num': 64, 'conv_size': 5, 'pool_size': 2, 'hidden_size': 1024, 'learning_rate': 1e-4, 'batch_num': 2000, 'batch_size': 32}
+    run_trial(params)
+```
+
+Note: If you want to see the full implementation, please refer to [examples/trials/mnist/mnist_before.py][7]
+
+The above code can only try one set of parameters at a time, if we want to tune learning rate, we need to manually tune the hyperparameters and start the trial again and again.
+
+NNI is born for helping user do the tuning jobs, the NNI working process is presented below:
+
+```
+input: search space, trial code, config file
+output: one optimal hyperparameter configuration
+
+1: For t = 0, 1, 2, ..., maxTrialNum,
+2:      hyperparameter = chose a set of parameter from search space
+3:      final result = run_trial_and_evaluate(hyperparameter)
+4:      report final result to NNI
+5:      If reach the upper limit time,
+6:          Stop the experiment
+7: return hyperparameter value with best final result
+```
+
+If you want to use NNI to automatically train your model and find the optimal hyper-parameters, you need to do three changes base on your code:
+
+**Three things required to do when using NNI**
+
+**Step 1**: Give a `Search Space` file in json, includes the `name` and the `distribution` (discrete valued or continuous valued) of all the hyperparameters you need to search.
+
+```diff
+-   params = {'data_dir': '/tmp/tensorflow/mnist/input_data', 'dropout_rate': 0.5, 'channel_1_num': 32, 'channel_2_num': 64,
+-   'conv_size': 5, 'pool_size': 2, 'hidden_size': 1024, 'learning_rate': 1e-4, 'batch_num': 2000, 'batch_size': 32}
++ {
++     "dropout_rate":{"_type":"uniform","_value":[0.5, 0.9]},
++     "conv_size":{"_type":"choice","_value":[2,3,5,7]},
++     "hidden_size":{"_type":"choice","_value":[124, 512, 1024]},
++     "batch_size": {"_type":"choice", "_value": [1, 4, 8, 16, 32]},
++     "learning_rate":{"_type":"choice","_value":[0.0001, 0.001, 0.01, 0.1]}
++ }
+```
+
+*Implemented code directory: [search_space.json][3]*
+
+**Step 2**: Modified your `Trial` file to get the hyperparameter set from NNI and report the final result to NNI.
+
+```diff
++ import nni
+
+  def run_trial(params):
+      mnist = input_data.read_data_sets(params['data_dir'], one_hot=True)
+
+      mnist_network = MnistNetwork(channel_1_num=params['channel_1_num'], channel_2_num=params['channel_2_num'], conv_size=params['conv_size'], hidden_size=params['hidden_size'], pool_size=params['pool_size'], learning_rate=params['learning_rate'])
+      mnist_network.build_network()
+
+      with tf.Session() as sess:
+          mnist_network.train(sess, mnist)
+          test_acc = mnist_network.evaluate(mnist)
+-         print ("Final result is", test_acc)
```

Comment:
original line is ok to keep to output in console. NNI doesn't require to remove it.

Reply:
Make sense. Done.